tools/tests/test_ptool.py

```diff
@@ -0,0 +1,45 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2019 Intel Corporation
+# All rights reserved.
+import sys
+import unittest
+import subprocess
+from unittest.mock import patch
+
+from tpm2_pyesys.util.simulator import SimulatorTest
+
+from tpm2_pkcs11 import tpm2_ptool as tool
+from tpm2_pkcs11.utils import TemporaryDirectory
+
+
+class TestPTool(SimulatorTest):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.abrmd = subprocess.Popen(
+            ["sudo", "-u", "tss", "tpm2-abrmd", "--tcti=mssim"]
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        for pid in [
+            line.split()[1]
+            for line in subprocess.check_output(["ps", "aux"]).decode().split("\n")
+            if "tpm2-abrmd" in line
+        ]:
+            subprocess.call(["sudo", "kill", pid])
```

Comment:
I hate myself for this but its the only way obi-wan

Reply:
I'm assuming that having to run as root or tss user has to do with dbus, or is that incorrect.

Ping: @flihp @AndreasFuchsSIT

Reply:
tpm2-abrmd needs to run as user tss/root, because /dev/tpm0 is tss:root and /dev/tpmrm0 is tss:tss.
If you're using mssim, you can however run tpm2-abrmd as $USER without sudo...

P.S. I'm reworking tpm2-pkcs11 atm which may void python-code in this project. Just as a heads-up...
tpm2-pyesys remains valid though...

Reply:
Ah okay great to know. Thank you!

Reply:
Step 1 is to not care about using an actual TPM device and just build all your testing for the simulator.